docs/docs/development/new-behavior.mdx

```diff
@@ -12,20 +12,46 @@ Before reading this section, it is **vital** that you read through our [clean ro
 
 ## Overview
 
-This document outlines how to develop a [behavior](../keymaps/behaviors/index.mdx) for ZMK and prepare the changes for a pull request.
+Behaviors refer to the actions that are invoked when a key is pressed or released.
+This guide outlines how to create a [ZMK module](../development/module-creation.md) that contains a new [behavior](../keymaps/behaviors/index.mdx).
 
-Behaviors are assigned to key positions and determine what happens when they are pressed and released. They are implemented in Zephyr as "devices": they consist of a devicetree binding file, which specifies the properties of the behavior, and a driver written in C code. This allows for the ability to create unique instances of these behaviors in [keymaps](../keymaps/index.mdx) or devicetree-source-include files (`.dtsi`). While instances of behaviors stored in keymaps are created by end-users for their personal needs, the instances that live in the .dtsi files are stored and documented in ZMK directly, which removes the need for end-users to set up common use-cases of these behaviors in their personal keymaps.
+:::info
+If an out-of-tree behavior's use-case is deemed widespread enough to be merged into upstream ZMK, a new pull request may be issued instead.
+While this page describes practices that maximize compatibility between module-focused development and pull-request-based workflows, we also note specific modifications required for the latter to work properly.
+:::
+
+In the context of the Zephyr RTOS, behaviors are implemented as "devices", which contain:
+
+- A devicetree binding file, which declare behavior's properties
+- A device driver written in C code
+- Optionally, devicetree-source-include files (`.dtsi`), which contain instances of the behavior that may be included directly in keymaps, rather than being explicitly defined in a `behaviors` node
 
 The general process for developing behaviors is:
 
-1. [Create the behavior](#creating-the-behavior)
-   1. [Create the devicetree binding (`.yaml`)](#creating-the-devicetree-binding-yaml)
-   1. [Create the driver (`.c`)](#creating-the-driver-c)
-   1. [Update `app/CmakeLists.txt` to include the new driver](#updating-appcmakeliststxt-to-include-the-new-driver)
-   1. [Define common use-cases for the behavior (`.dtsi`) (Optional)](#defining-common-use-cases-for-the-behavior-dtsi-optional)
+1. [Create a new behavior repository](#creating-a-new-behavior-repository)
```

Comment:
I think it might be better to just leave the top-level items here, what do you think? This looks too much like a ToC.